src/merkle_tree.rs

```diff
@@ -1,6 +1,7 @@
 use sha3::{Digest, Sha3_256};
 
 type HashedData = [u8; 32];
+type Position = usize;
```

Comment:
Is this needed? I think `usize` is clear enough.

Reply:
You're right, that's a bit un-necessary.